openverse_catalog/dags/providers/provider_api_scripts/freesound.py

```diff
@@ -8,294 +8,249 @@
+        set_url = media_data.get("pack")
+        if set_url is not None:
+            set_id, set_name = self._get_set_info(set_url)
+            return set_id, set_name, set_url
+        else:
+            return None, None, None
+
+    @staticmethod
+    def _get_preview_filedata(preview_type, preview_url):
+        return {
+            "url": preview_url,
+            "filetype": preview_type.split("-")[-1],
+            "bit_rate": FreesoundDataIngester.preview_bitrates[preview_type],
+        }
 
-def _get_audio_set_info(media_data):
-    # set id, set name, set url
-    set_url = media_data.get("pack")
-    if set_url is not None:
-        set_id, set_name = _get_set_info(set_url)
-        return set_id, set_name, set_url
-    else:
-        return None, None, None
-
-
-@functools.lru_cache(maxsize=1024)
-def _get_set_info(set_url):
-    response_json = delayed_requester.get_response_json(
-        set_url,
-        3,
-        query_params={"token": API_KEY},
-    )
-    set_id = response_json.get("id")
-    set_name = response_json.get("name")
-    return set_id, set_name
-
-
-def _get_preview_filedata(preview_type, preview_url):
-    return {
-        "url": preview_url,
-        "filetype": preview_type.split("-")[-1],
-        "bit_rate": preview_bitrates[preview_type],
-    }
+    @retry(flaky_exceptions, tries=3, delay=1, backoff=2)
+    def _get_audio_file_size(self, url):
+        """
+        Get the content length of a provided URL.
+        Freesound can be finicky, so we want to retry it a few times on
+        these conditions:
+          * SSLError - 'EOF occurred in violation of protocol (_ssl.c:1129)'
+          * ConnectionError - '[Errno 113] No route to host'
+
+        Both of these seem transient and may be the result of some odd behavior on the
+        Freesound API end. We have an API key that's supposed to be maxed out, so
+        I can't imagine it's throttling (aetherunbound).
+
+        TODO(obulat): move filesize detection to the polite crawler
+        """
+        return int(requests.head(url).headers["content-length"])
+
+    def _get_audio_files(self, media_data):
+        # This is the original file, needs auth for downloading.
+        # bit_rate in kilobytes, converted to bytes
+        alt_files = [
+            {
+                "url": media_data.get("download"),
+                "bit_rate": int(media_data.get("bitrate")) * 1000,
+                "sample_rate": int(media_data.get("samplerate")),
+                "filetype": media_data.get("type"),
+                "filesize": media_data.get("filesize"),
+            }
+        ]
+        previews = media_data.get("previews")
+        # If there are no previews, then we will not be able to play the file
+        if not previews:
+            return None
+        main_file = self._get_preview_filedata(
+            "preview-hq-mp3", previews["preview-hq-mp3"]
+        )
+        main_file["audio_url"] = main_file.pop("url")
```

Comment:
I can see this was in the old code, but I'm not sure why `_get_preview_filedata` doesn't just name `url` `audio_url`?

Reply:
I'm not 100% sure, but I think it's because the URL field is called `url` for `alt_files`, and `audio_url` for the main audio in the AudioStore. And `_get_preview_filedata` is mainly used for populating the `alt_files`, not for the `main` file.

Reply:
It's definitely `audio_url` in the `AudioStore` for the main file:

https://github.com/WordPress/openverse-catalog/blob/4a9c008a150b2fc5a88b8f9c64f5bd5beef66ac9/openverse_catalog/dags/common/storage/audio.py#L46

I checked the database and it does look like we're using `url` in the `alt_files` dict:
```
deploy@localhost:openledger> select url, alt_files from audio where provider='freesound' limit 1;
-[ RECORD 1 ]-------------------------
url       | https://freesound.org/data/previews/100/100041_1578278-hq.mp3
alt_files | [{"url": "https://freesound.org/apiv2/sounds/100041/download/", "bit_rate": "0", "filesize": "5056680", "filetype": "wav", "sample_rate": "44100"}]
SELECT 1
```

Reply:
I can see that the AudioStore expects `audio_url`, and that `alt_files` uses `url` (notably the `audio_url` also gets converted to `url` for the main file in the actual DB which can be a little confusing, I opened #784 for it).

But it looks like `_get_preview_filedata` is only ever used once, to get the url and other data for the main file? I don't see why it couldn't just return the key as `audio_url` so it doesn't have to get renamed here. Unless I'm missing another place where it's being used.

Reply:
OH, I see what you're saying now, apologies! I'm not sure it's necessary either, I'll rework this a bit so it's clearer!